addon/ng2/tasks/uninstall.js

```diff
@@ -0,0 +1,157 @@
+/* jshint node: true, esnext: true */
+'use strict';
+
+const Promise     = require('ember-cli/lib/ext/promise');
+const Task        = require('ember-cli/lib/models/task');
+const chalk       = require('chalk');
+const path        = require('path');
+const fs          = require('fs');
+const fse         = require('fs-extra');
+const exec        = Promise.denodeify(require('child_process').exec);
+const packageJSON = path.resolve(process.cwd(), 'package.json');
+const nodeModules = path.resolve(process.cwd(), 'node_modules');
+
+module.exports = Task.extend({
+  command: '',
+  completionOKMessage: 'Successfully uninstalled.',
+  completionErrorMessage: 'Error uninstalling package.',
+
+  run: function(options) {
+    this.package = options.package;
+    this.typings = options.typings;
+
+    this.disableLogger();
+
+    this.ui.startProgress(chalk.green('Uninstalling package:', chalk.yellow(this.package)), chalk.green('.'));
+
+    return this.uninstallProcedure();
+  },
+
+  uninstallProcedure: function() {
+    const that = this;
+
+    return that.removeFromPackageJSON(that.package)
```

Comment:
The goal of promises was to avoid this kind of thing :)  you can simplify this like this:

```typescript
    return that.removeFromPackageJSON(that.package)
      .then(() => that.npmUninstall(that.package))
      .then(() => that.unlinkPackage(that.package))
      .then(() => that.removeFromSystemJSConfig(that.package))
      .then(() => {
        that.ui.stopProgress();
        if (that.typings) {
          that.ui.startProgress(chalk.green('Uninstalling typings:', chalk.yellow(that.typings)), chalk.green('.'));
          return that.typingsUninstall(that.typings)
        }
      })
      .then(() => {
        that.ui.stopProgress();
        that.announceOKCompletion();
      });
```